src/interpreter/validation.rs

```diff
@@ -0,0 +1,85 @@
+use crate::error::InterpreterError;
+use crate::interpreter::Interpreter;
+use fuel_asm::PanicReason;
+use fuel_tx::{Input, Output, Transaction};
+use fuel_types::{AssetId, Word};
+use std::collections::HashMap;
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct CheckedTransaction {
```

Comment:
Looks like this code belongs to `fuel-tx` rather than `fuel-vm`. In fact, we already have a data structure to contain transaction [Metadata](https://docs.rs/fuel-tx/0.6.0/fuel_tx/struct.Metadata.html).

I believe we should insert this balances map there.

Reply:
The reason I don't think this checked transaction wrapper type should be outside of the VM is based on my proposal in the related issue:

> We could make it so the only way to construct these wrapped transactions is by running the validation method of the VM. Then the interpreter init methods could be modified to only accept these checked transactions.

If CheckedTransaction was in a different crate, it would hold no significance or guarantees for the VM. Anyone could spoof a CheckedTransaction and lead to undefined behavior in the VM.

Reply:
based on your comment here: https://github.com/FuelLabs/fuel-vm/pull/93#pullrequestreview-921424472

I'm coming around to the idea that we could actually move all validations the VM needs into fuel-tx, and that this isn't needed anymore. I'm going to close this PR and focus on cleaning up the validation tech debt.